### ISSUE TITLE
Forgotten password (API)

### ISSUE THREAD

**Issue by ljwagerfield** [fifthweek/fifthweek-web#103]
Acceptance criteria:
- New service endpoint:
  - Accept email or username and always return a success message.
  - Email notification sent to user with link to change password.
  - Links to page that asks user to enter new password.
- New service endpoint:
  - Accept new password and ciphertext previously generated by endpoint.
  - Change password if ciphertext valid.
  - Email notification sent to user asserting their password has been changed.

See:
http://www.asp.net/identity/overview/features-api/account-confirmation-and-password-recovery-with-aspnet-identity

**Comment by ljwagerfield** [comment]
Added a DTO for username/email in this commit:
c9f74a33509fb0718806249c75bc8604a9f541a4

**Comment by ljwagerfield** [comment]
Also adding an endpoint for checking if tokens are still valid. 

This will be helpful when the user hits the page: we can check token validity in the background, and flag an error to the user if the token has expired.

Unfortunately I had to do a *minor* hack to get this working: there's a private string constant in the implementation of ASP.NET's UserManager class that I've had to repeat in our code. It's like a salt that's applied to tokens, to discriminate that this token is specifically intended for password resets, as opposed to email address confirmation, etc. AFAIK there isn't a way around it :(
